### PR TITLE
Fetch source and automation data for a given application

### LINKED ISSUE #491: Store references to Kustomization and Source k8s objects
When we `add` an app, we will need to save a reference to the flux Kustomization/HelmRelease and Source objects that are related. This will allow us to draw the reconciliation graph later.

After chatting with Flux maintainers, the best way to express the relationship between `Application` and `Kustomization | HelmRelease` and `Source` is via an `ownerReference`.

Note that this has garbage collection implications. K8s documentation on `ownerReference`:
https://kubernetes.io/docs/concepts/workloads/controllers/garbage-collection/

https://weaveworks.slack.com/archives/C0CG4EP6F/p1626710106321700

Implementation notes:
* I would imagine we would bake in an owner reference to the `KustomizationSpec |SourceSpec` when we create them

### PR DESCRIPTION
**Pull request by jpellizzari**
Closes #491 

In order to retrieve the sources and automatons for a given `Application`, we need to know some things about the related objects. We assume that a source and automation are named the same as the application.

A temporary UI to demonstrate:
Screenshot from 2021-07-21 07-58-20

### CODE REVIEW

**Review by josecordaz — APPROVED** [review]
LGTM

**Review by palemtnrider — COMMENTED** [review]
Mainly minors changes to increase clarity.

File: pkg/server/server.go
Comment (jpellizzari) [reply pkg/server/server.go:70]: @palemtnrider `Deployment` is already a word used in Kubernetes. Do we not already call these things "GOAT" for "GitOps Automation" (or at least that is what I thought it was called)?

**Review by luizbafilho — APPROVED** [review]

File: api/applications/applications.proto
Comment (palemtnrider) [inline api/applications/applications.proto:48]: Can you elaborate on using string as the type instead of timestamp?  Having timestamp instead of strings makes it easier for non-human interfaces to consume. Also, a UI will have an easier time converting to local time when displaying to the user.

File: api/applications/applications.proto
Comment (palemtnrider) [inline api/applications/applications.proto:56]: the CRD uses the term `deployment` instead of `automation`

File: pkg/server/server.go
Comment (palemtnrider) [reply pkg/server/server.go:59]: I don't see the app name being added to the error here or in findFluxObject?

File: pkg/server/server.go
Comment (palemtnrider) [inline pkg/server/server.go:111]: It's possible the only one of the objects exists when findFluxObjects is called and I think that should be OK.  If only the source object is present, we should still return it and vice versa for the deployment type.  In fact, neither may exist in the cluster yet and that should be OK.

File: pkg/services/app/app.go
Comment (palemtnrider) [inline pkg/services/app/app.go:19]: we should use the ones from application_types here instead of defining them again.

**Review by palemtnrider — CHANGES_REQUESTED** [review]
Looking go so far.  I suggested a few items I'd like to see cleaned up before we merge.

File: api/applications/applications.proto
Comment (jpellizzari) [reply api/applications/applications.proto:48]: Looks like I used an `int32` in the Flux UI:
https://github.com/fluxcd/webui/blob/f48d23e5095a04949ad276fc6da1349f380df457/pkg/clustersserver/clustersserver.go#L712

Which comes back in seconds. Hopefully enough resolution?:
https://github.com/fluxcd/webui/blob/f48d23e5095a04949ad276fc6da1349f380df457/ui/pages/Events.tsx#L34

Will fix!

File: pkg/services/app/app.go
Comment (jpellizzari) [reply pkg/services/app/app.go:19]: See the comment on the line below. It would muddy up the intent of this PR. Follow up issue created.

File: api/applications/applications.proto
Comment (jpellizzari) [reply api/applications/applications.proto:56]: From another comment that may have shown as "outdated":

> `Deployment` is already a word used in Kubernetes. Do we not already call these things "GOAT" for "GitOps Automation" (or at least that is what I thought it was called)?

So I think we should discuss changing that command-line flag for this reason ^^. That would be outside of the scope of this PR, so I will create a follow up and change this to be consistent for now.

File: pkg/server/server.go
Comment (jpellizzari) [reply pkg/server/server.go:129]: See above re `Deployment`

File: api/applications/applications.proto
Comment (jpellizzari) [reply api/applications/applications.proto]: Changed to `int32`

File: pkg/server/server.go
Comment (jpellizzari) [reply pkg/server/server.go:128]: Changed to return `nil` with no error in the event of a "NotFound" error from kubernetes

File: api/applications/applications.proto
Comment (jpellizzari) [reply api/applications/applications.proto]: Changed to `deployment_conditions`

File: pkg/server/server.go
Comment (jpellizzari) [reply pkg/server/server.go]: See the second review in this PR for further discussion

**Review by palemtnrider — APPROVED** [review]

**Pull request merged**
